travis.sh

```diff
@@ -2,43 +2,70 @@
 
 set -uexo pipefail
 
-DIGGER_DIR="../digger"
-DIGGER="../digger/digger"
+TRAVIS_BRANCH=${TRAVIS_BRANCH:-master}
+DMD="../dmd/src/dmd"
+N=2
 
 # set to 64-bit by default
 if [ -z ${MODEL:-} ] ; then
     MODEL=64
 fi
 
+clone() {
+    local url="$1"
+    local path="$2"
+    local branch="$3"
+    for i in {0..4}; do
+        if git clone --depth=1 --branch "$branch" "$url" "$path"; then
+            break
+        elif [ $i -lt 4 ]; then
+            sleep $((1 << $i))
+        else
+            echo "Failed to clone: ${url}"
+            exit 1
+        fi
+    done
+}
+
 test_rdmd() {
     # run rdmd internal tests
-    rdmd -m$MODEL -main -unittest rdmd.d
+    rdmd --compiler=$DMD -m$MODEL -main -unittest rdmd.d
 
     # compile rdmd & testsuite
-    dmd -m$MODEL rdmd.d
-    dmd -m$MODEL rdmd_test.d
+    $DMD -m$MODEL rdmd.d
+    $DMD -m$MODEL rdmd_test.d
 
     # run rdmd testsuite
-    ./rdmd_test
+    ./rdmd_test --compiler=$DMD
 }
 
-build_digger() {
-    git clone --recursive https://github.com/CyberShadow/Digger "$DIGGER_DIR"
-    (cd "$DIGGER_DIR" && rdmd --build-only -debug digger)
-}
+setup_repos()
+{
+    for repo in dmd druntime phobos dlang.org ; do
+        if [ ! -d "../${repo}" ] ; then
+            if [ $TRAVIS_BRANCH != master ] && [ $TRAVIS_BRANCH != stable ] &&
+                   ! git ls-remote --exit-code --heads https://github.com/dlang/$proj.git $TRAVIS_BRANCH > /dev/null; then
+                # use master as fallback for other repos to test feature branches
+                clone https://github.com/dlang/${repo}.git ../${repo} master
+            else
+                clone https://github.com/dlang/${repo}.git ../${repo} $TRAVIS_BRANCH
+            fi
+        fi
+    done
 
-install_digger() {
-    $DIGGER build --model=$MODEL "master"
-    export PATH=$PWD/result/bin:$PATH
+    make -j$N -C ../dmd/src -f posix.mak MODEL=$MODEL HOST_DMD=dmd all
```

Comment:
Missing `make -j$N -C src -f posix.mak MODEL=$MODEL HOST_DMD=$DMD dmd.conf` after that, hence the failure with rdmd.